tests/src/python/test_qgsfeatureiterator.py

```diff
@@ -494,21 +504,29 @@ def callback(feature):
         res = [f['x'] for f in
                layer.getFeatures(QgsFeatureRequest().setInvalidGeometryCheck(QgsFeatureRequest.GeometryAbortOnInvalid))]
         self.assertEqual(res, ['a'])
+        res = [f for f in
+               layer.getFeatures(QgsFeatureRequest().setInvalidGeometryCheck(QgsFeatureRequest.GeometryFixInvalid))]
+        self.assertEqual(set([f['x'] for f in res]), {'a', 'b', 'c', 'd'})
+        self.assertNotEqual(res[3].geometry().asWkt(), f4.geometry().asWkt())
 
         # check with features with changed geometry
         layer.rollBack()
         layer.startEditing()
         layer.changeGeometry(2, QgsGeometry.fromWkt('Polygon((0 0, 1 0, 1 1, 0 1, 0 0))'))  # valid
         layer.changeGeometry(3, QgsGeometry.fromWkt('Polygon((0 0, 1 0, 0 1, 1 1, 0 0))'))  # invalid
-        res = [f['x'] for f in
-               layer.getFeatures(QgsFeatureRequest().setInvalidGeometryCheck(QgsFeatureRequest.GeometryNoCheck))]
-        self.assertEqual(set(res), {'a', 'b', 'c'})
         res = [f['x'] for f in
                layer.getFeatures(QgsFeatureRequest().setInvalidGeometryCheck(QgsFeatureRequest.GeometrySkipInvalid))]
         self.assertEqual(set(res), {'a', 'b'})
         res = [f['x'] for f in
                layer.getFeatures(QgsFeatureRequest().setInvalidGeometryCheck(QgsFeatureRequest.GeometryAbortOnInvalid))]
-        self.assertEqual(res, ['a', 'b'])
+        self.assertEqual(set(res), {'a', 'b'})
+        res = [f for f in
+               layer.getFeatures(QgsFeatureRequest().setInvalidGeometryCheck(QgsFeatureRequest.GeometryNoCheck))]
+        self.assertEqual(set([f['x'] for f in res]), {'a', 'b', 'c'})
```

Comment:
same here -- I can't see the motivation behind changing this test and am concerned that you've done this to mask an issue introduced against the older version of the test

Reply:
No, I did this to reuse the test result to compare the resulting geometries. I could leave the old one and the new one but this would test the same thing twice and one would only aim to keep the results to compare with the autofix approach.

Reply:
@nyalldawson Ok to mark the test changes as resolved?